core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly

```diff
@@ -57,7 +57,7 @@ THE SOFTWARE.
                 <p>
                   ${%Or if the slave is headless:}
                 </p>
-                <pre>java -jar <a href="${rootURL}/jnlpJars/slave.jar">slave.jar</a> -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp</pre>
+                <pre>java ${it.launcher.vmargs}${it.launcher.vmargs==null?'':' '}-jar <a href="${rootURL}/jnlpJars/slave.jar">slave.jar</a> -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp</pre>
```

Comment:
Less confusing to write

    ${it.launcher.vmargs == null ? '' : ' ' + it.launcher.vmargs}

And did you not want a trailing space before `-jar`?